### ISSUE TITLE
Customize person browse & facets

### ISSUE THREAD

**Issue by nathangibson** [usaybia/srophe-eXist-app#14]
Waiting on #12

**Comment by nathangibson** [comment]
Trying to figure out what I need to do to turn facets on. E.g., there should be a bunch of records now with `state[@type="occupation"]` that can be faceted on `@role`

Does this look correct? https://github.com/usaybia/srophe-eXist-app/pull/28

**Comment by wsalesky** [comment]
@nathangibson yes that looks correct. You will need to rerun modules/index.xql to register the new facet with Lucene. I will run it today.

**Comment by wsalesky** [comment]
@nathangibson the tools/ directory in usaybia-data is causing errors with the indexer. Any chance you can pull those out? Or do they need to be loaded in eXist? If they are not part of the data application, then I would store them in a different repository. I will do a rebuild of the data without the tools and install it, but the webhooks will add them back in anytime you make a change to them.

**Comment by wsalesky** [comment]
@nathangibson This has been updated. There are some changes to how facets are handled to take advantage of the new Lucene supported facets. I have found these a little touchier than the original, but also faster, so worth the effort. 

You should add a nam and label to the facet, see: https://github.com/usaybia/srophe-eXist-app/blob/master/persons/facet-def.xml#L3

If the facet needs any additional processing to get the correct result you will need to make a custom function here: https://github.com/usaybia/srophe-eXist-app/blob/master/modules/lib/facets.xql#L193

And then the .xconf file will need to be rebuilt and the data reindexed. You can do this using the the modules/index.xql file.

**Comment by nathangibson** [comment]
I ran the modules/index.xql using exide on the server, but I don't see any change. Still no facets listed.

**Comment by wsalesky** [comment]
Interesting. I will have to troubleshoot. If you open the .xconf file in /db/system/config/usaybia-data/ and then save it (with out editing) this also triggers a reindex and this actually works. So there must be something going on in my index script that is causing it to fail. I re-indexed and the occupation facet is there. I will try to sort out what is going on with the index.xql, may not happen this week though.

**Comment by nathangibson** [comment]
Thanks this is looking great! And no problem if you can't get the index.xql sorted out this week. For this week I just wanted to have some sort of facet available to demo.

**Comment by nathangibson** [comment]
Only the list is not ordered as I would expect: it looks alphabetical, but I would have expected the line `<order-by direction="ascending">count</order-by>` to order it by number of matching records??

**Comment by wsalesky** [comment]
@nathangibson This should be fixed. Pending webhooks.
